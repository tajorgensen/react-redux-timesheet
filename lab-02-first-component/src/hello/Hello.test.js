import React from 'react';
import { shallow } from 'enzyme';

import Hello from './Hello';

describe('Hellow World:', () => {

    it ('renders without exploding', () => {
        expect(shallow(<Hello />)).toHaveLength(1);
    });

    it ('should render with default text', () => {
        const component = shallow(<Hello />);

        expect(component).toIncludeText('Howdy');
        expect(component).toIncludeText('Partner');
    });

    it ('should render with our props', () => {
        const component = shallow(<Hello friend="Fred" />);

        expect(component).toIncludeText('Howdy');
        expect(component).toIncludeText('Fred');
        expect(component).not.toIncludeText('Partner');
    })

});