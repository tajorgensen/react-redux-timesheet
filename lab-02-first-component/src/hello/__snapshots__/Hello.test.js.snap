// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hellow World: should render to match the snapshot 1`] = `
<div
  className="hello"
>
  <h1>
    Howdy!!
  </h1>
  <h1>
    Wyatt
  </h1>
  <p>
    Congratulations!  You have created your first React component
  </p>
</div>
`;
